packages/contracts-periphery/deploy/nft-bridge/L1ERC721BridgeImplementation.ts

```diff
@@ -1,15 +1,96 @@
 /* Imports: External */
 import { DeployFunction } from 'hardhat-deploy/dist/types'
-import { ethers } from 'hardhat'
+import '@eth-optimism/hardhat-deploy-config'
+import '@nomiclabs/hardhat-ethers'
+import 'hardhat-deploy'
 
 const deployFn: DeployFunction = async (hre) => {
   const { deployer } = await hre.getNamedAccounts()
+  const { deploy } = hre.deployments
+  const { getAddress } = hre.ethers.utils
 
-  await hre.deployments.deploy('L1ERC721Bridge', {
+  const Deployment__L1ERC721BridgeProxy = await hre.deployments.get(
+    'L1ERC721BridgeProxy'
+  )
+
+  const L1ERC721BridgeProxy = await hre.ethers.getContractAt(
+    'Proxy',
+    Deployment__L1ERC721BridgeProxy.address
+  )
+  const admin = await L1ERC721BridgeProxy.admin()
+  if (getAddress(admin) !== getAddress(deployer)) {
+    throw new Error('deployer is not proxy admin')
+  }
+
+  // Get the address of the currently deployed L1CrossDomainMessenger.
+  // This should be the address of the proxy
+  const Deployment__L1CrossDomainMessenger = await hre.deployments.get(
+    'Proxy__OVM_L1CrossDomainMessenger'
+  )
+  const L1CrossDomainMessengerAddress =
+    Deployment__L1CrossDomainMessenger.address
+
+  const predeploy = '0x4200000000000000000000000000000000000014'
+  // Deploy the L1ERC721Bridge. The arguments are
+  // - messenger
+  // - otherBridge
+  // Since this is the L1ERC721Bridge, the otherBridge is the
+  // predeploy address
+  await deploy('L1ERC721Bridge', {
```

Comment:
I'd like it if we can postfix this with Impl to remove ambiguity.

Reply:
We don't do this with `contracts-bedrock`, would prefer to be similar to that, then when using hardhat to do things like `hre.artifacts.readArtifact` you will need to know if it has the impl suffix or not